### ISSUE TITLE
git push takes a long time, even when only changing a single page

### ISSUE THREAD

**Issue by rohieb** [Git-Mediawiki/Git-Mediawiki#36]
I have cloned a wiki with ~500 pages, with by_ref strategy. Now when I change a page, commit and push, the following happens:

```
Last local mediawiki revision found is 13613.
Logged in mediawiki user "rohieb".
Listing pages on remote wiki...
527 pages found.
Getting last revision id on tracked pages...
Last remote revision found is 13613.
[...]
```

The process of getting revision IDs takes about 5 minutes, from the network activity I guess that every single page revision is queried on the API. However in my case, my commit changes only one page, so I would think only one page revision needs to be checked.

I tried setting the fetch strategy to `by_page` as well as setting `remote.mywiki.dumbPush`, but that does not change the behaviour.

**Comment by moy** [comment]
The fetch strategy doesn't impact push. Push strategies is not implemented (see #6, which you already know about actually).

**Comment by rohieb** [comment]
On 14.07.2016 10:02, Matthieu Moy wrote:
> (see #6
> <https://github.com/Git-Mediawiki/Git-Mediawiki/issues/6>, which you
> already know about actually).

You're right, sorry about the noise.

 - Roland
